### PR TITLE
Parquet Cleanup

### PR DESCRIPTION
**Pull request by zhenxiao**
@dain @electrum @nezihyigitbasi now New Parquet Reader is production ready
this patch is to remove the old Parquet reader, and a cleanup of Parquet code

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java [hunk 1]
```diff
@@ -227,4 +233,38 @@ public static ParquetPageSource createParquetPageSource(
             throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, message, e);
         }
     }
+
+    public static TupleDomain<ColumnDescriptor> getParquetTupleDomain(Map<List<String>, RichColumnDescriptor> descriptorsByPath, TupleDomain<HiveColumnHandle> effectivePredicate)
```
Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java]: Any reason why you moved these methods from the `PredicateUtils` class to here?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java [hunk 2]
```diff
@@ -333,43 +321,6 @@ public void testAvro(int rowCount)
                 .collect(toList());
     }
 
-    @Test(dataProvider = "rowCount")
```
Comment (nezihyigitbasi) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: What's the reason for removing these tests?

File: presto-parquet/src/main/java/com/facebook/presto/parquet/ParquetTypeUtils.java [hunk 3]
```diff
@@ -256,4 +243,20 @@ public static boolean isValueNull(boolean required, int definitionLevel, int max
     {
         return !required && (definitionLevel == maxDefinitionLevel - 1);
     }
+
+    public static long getShortDecimalValue(byte[] bytes)
```
Comment (nezihyigitbasi) [inline presto-parquet/src/main/java/com/facebook/presto/parquet/ParquetTypeUtils.java]: Is this copied from the hive module? If so, please add a comment that this is copied.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java (hunk 2 above)
Comment (zhenxiao) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: these tests are for old Parquet reader only

File: presto-parquet/src/main/java/com/facebook/presto/parquet/ParquetTypeUtils.java (hunk 3 above)
Comment (zhenxiao) [reply presto-parquet/src/main/java/com/facebook/presto/parquet/ParquetTypeUtils.java]: yes, will do

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java (hunk 1 above)
Comment (zhenxiao) [reply presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java]: it is mostly just called here. Try to not have cross module dependency

**Comment by zhenxiao** [comment]
@nezihyigitbasi separate into 2 commits: one for remove old parquet reader, one for new Parquet reader code refactor

**Comment by zhenxiao** [comment]
thank you @nezihyigitbasi 
get it updated

**Comment by zhenxiao** [comment]
kindly ping @nezihyigitbasi

**Comment by nezihyigitbasi** [comment]
@zhenxiao can you please rebase? I plan to review this one in the next few days.

**Comment by zhenxiao** [comment]
get it rebased. @nezihyigitbasi

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java (hunk 1 above)
Comment (nezihyigitbasi) [reply presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java]: This module already depends on `presto-parquet`, also this class already imports `PredicateUtils` methods. I think we can keep these methods in their original class, the `PredicateUtils`.

File: presto-parquet/pom.xml [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+        <dependency>
+            <groupId>it.unimi.dsi</groupId>
+            <artifactId>fastutil</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.xerial.snappy</groupId>
+            <artifactId>snappy-java</artifactId>
+            <version>1.0.5</version>
```
Comment (nezihyigitbasi) [inline presto-parquet/pom.xml]: In the top level project pom we use `1.1.1.7` version, why do we override that here?

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java (hunk 1 above)
Comment (zhenxiao) [reply presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java]: oh, just found, HiveColumnHandle could not include in presto-parquet module, so I leave it here.

**Comment by zhenxiao** [comment]
thank you, @nezihyigitbasi 
comments addressed

**Comment by zhenxiao** [comment]
kindly ping

**Comment by zhenxiao** [comment]
kindly ping

**Comment by zhenxiao** [comment]
kindly ping

**Comment by zhenxiao** [comment]
@nezihyigitbasi kindly ping :)

**Comment by haozhun** [comment]
@nezihyigitbasi is out for an extended period of time.

**Comment by zhenxiao** [comment]
oh, get it. thank you @haozhun @dain are you free to review?

File: pom.xml [hunk 5]
```diff
@@ -77,6 +77,7 @@
         <module>presto-blackhole</module>
         <module>presto-memory</module>
         <module>presto-orc</module>
+        <module>presto-parquet</module>
```
Comment (dain) [inline pom.xml]: Can you retitle this commit "Move Parquet reader to new presto-parquet module"?

File: presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/Predicate.java [hunk 6]
```diff
@@ -29,14 +29,14 @@ public boolean matches(long numberOfRows, Map<ColumnDescriptor, Statistics<?>> s
...
-        public boolean matches(Map<ColumnDescriptor, ParquetDictionaryDescriptor> dictionaries)
+        public boolean matches(Map<ColumnDescriptor, DictionaryDescriptor> dictionaries)
         {
             return true;
         }
     };
 
     /**
-     * Should the Parquet Reader process a file section with the specified statistics.
+     * Should the  Reader process a file section with the specified statistics.
```
Comment (dain) [inline presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/Predicate.java]: The javadocs here have two spaces where `Parquet` was removed.

**Comment by zhenxiao** [comment]
@dain thank you, I get comments addressed
https://github.com/prestodb/presto/issues/10844 should be fixed in 203+ version

**Comment by zhenxiao** [comment]
kindly ping @nezihyigitbasi @dain

**Comment by dain** [comment]
It looks good to me.  @nezihyigitbasi do you have any additional concerns?

**Comment by nezihyigitbasi** [comment]
Merged, thanks @zhenxiao.

**Pull request merged**